tests/unit/lazy-video-providers-test.js

```diff
@@ -19,28 +19,28 @@ module('Lazy Video Providers', {
   }
 });
 
-test('_getProvider with explicit provider', function() {
+test('_getProvider based on `url`', function() {
   var provider;
 
-  provider = service._getProvider('youtube');
+  provider = service._getProvider('https://www.youtube.com/watch?v=gvdf5n-zI14');
```

Comment:
Could we add more test cases, particularly for the optional params YouTube sometimes adds to URLs?